bindings/java/mongocrypt/build.gradle.kts

```diff
@@ -254,41 +259,47 @@ tasks.javadoc {
     }
 }
 
-tasks.register("publishSnapshots") {
-    group = "publishing"
-    description = "Publishes snapshots to Sonatype"
-    if (version.toString().endsWith("-SNAPSHOT")) {
-        dependsOn("downloadJnaLibs")
-        dependsOn(tasks.withType<PublishToMavenRepository>())
-        tasks.withType<PublishToMavenRepository>().forEach { t -> t.mustRunAfter("downloadJnaLibs") }
-    }
-}
-
-tasks.register("publishArchives") {
+tasks.register("publishToSonatype") {
     group = "publishing"
-    description = "Publishes a release and uploads to Sonatype / Maven Central"
+    description = """Publishes to Sonatype.
+        |
+        | - If the version string ends with SNAPSHOT then publishes to the Snapshots repo.
+        |   Note: Uses the JNA libs from the current build.
+        |
+        | - If is a release then publishes the release to maven central staging.
+        |   A release is when the current git tag is prefixed with java (eg: java-1.7.0)
+        |   AND the git tag version matches the version the build.gradle.kts.
+        |   Note: Uses the JNA libs from the associated tag.
+        |   Eg: Tag java-1.7.0 will use the JNA libs created by the 1.7.0 release tag.
+        |
+        | To override the JNA library downloaded use -DgitRevision=<git hash>
+    """.trimMargin()
+    val isRelease = version.toString().endsWith("-SNAPSHOT") || (isJavaTag && gitVersion == version)
```

Comment:
If version ends with `-SNAPSHOT` (e.g. `1.7.0-SNAPSHOT`) and mismatches tag (e.g. `java-1.7.1`) there is no error.

Consider expanding this check, and renaming to `versionOK`:

```kotlin
    val isSnapshot = version.toString().endsWith("-SNAPSHOT")
    var versionOK = false
    if (isSnapshot && !isJavaTag) {
        // This is a snapshot commit, and the commit is untagged.
        versionOK = true
    } else if (!isSnapshot && isJavaTag && gitVersion == version) {
        // This is a release commit, and the tag matches.
        versionOK = true
    }
```

Reply:
> If version ends with -SNAPSHOT (e.g. 1.7.0-SNAPSHOT) and mismatches tag (e.g. java-1.7.1) there is no error.

Like that now made it an error in that scenario.

I've kept it as not an error if its not a java tag though. As this task is run on all tags.